driver-async/src/test/functional/com/mongodb/async/client/RetryableWritesTest.java

```diff
@@ -153,24 +154,29 @@ public void shouldPassAllOutcomes() {
 
         BsonDocument result = new BsonDocument();
         boolean wasException = false;
+        Exception raisedException = null;
         try {
             result = helper.getOperationResults(operation);
         } catch (Exception e) {
             wasException = true;
-        }
-
-        if (outcome.containsKey("collection")) {
-            FutureResultCallback<List<BsonDocument>> futureResultCallback = new FutureResultCallback<List<BsonDocument>>();
-            collection.withDocumentClass(BsonDocument.class).find().into(new ArrayList<BsonDocument>(), futureResultCallback);
-            assertEquals(outcome.getDocument("collection").getArray("data").getValues(), futureResult(futureResultCallback));
+            raisedException = e;
         }
 
         if (outcome.getBoolean("error", BsonBoolean.FALSE).getValue()) {
             assertEquals(outcome.containsKey("error"), wasException);
+        } else if (wasException) {
+            fail("Unexpected exception: " + raisedException);
         } else {
             BsonDocument fixedExpectedResult = outcome.getDocument("result", new BsonDocument());
             assertEquals(fixedExpectedResult, result.getDocument("result", new BsonDocument()));
         }
+
+        if (outcome.containsKey("collection")) {
```

Comment:
Made this change when debugging this test, but seems generally useful to check for exceptions before checking the collection contents

Reply:
Ack